torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -610,8 +612,37 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
             "All tensor operands to scatter/gather must have the same size");
       }
     }
-    // Flatten the tensors (from all ranks) into a single big tensor.
-    flattened[i] = newLikeFlat(tensor_lists, i);
+
+    if (no_copy) {
+      // no_copy operation requires all tensors in tensor_list are contiguous views
+      // into a single flattened tensor
+      for (auto j = size_t{}; j < tensor_lists[i].size(); ++j) {
+        auto t = tensor_lists[i][j];
+        if (!tensor_lists[i][j].storage().is_alias_of(tensor_lists[i][0].storage()) ||
+            tensor_lists[i][j].storage_offset() != (tensor_lists[i][0].storage_offset() +
+              j * tensor_lists[i][0].numel())) {
+          no_copy = false;
```

Comment:
I wonder if it would be better to be explicit here and throw or at least warn if user specified no_copy but we can't actually do it?